pkg/metastore/badger.go

```diff
@@ -481,6 +485,39 @@ func (m *BadgerMetastore) GetOrCreateStacktraces(ctx context.Context, r *pb.GetO
 		stacktraceKeys = append(stacktraceKeys, MakeStacktraceKey(stacktrace))
 	}
 
+	const maxRetries = 2
+	var result retryableGetOrCreateStacktraces
+	var err error
+
+	m.logger.Debugf("GetOrCreateStacktraces with %d stacktraces", len(r.Stacktraces))
+	for i := 0; i < maxRetries; i++ {
+		result, err = m.retryableGetOrCreateStacktraces(r, stacktraceKeys)
+		if err != nil {
+			return res, err
+		}
+
+		res.Stacktraces = append(res.Stacktraces, result.stackTraces...)
+		if len(result.retryWith) == 0 {
+			break
+		}
+		stacktraceKeys = result.retryWith
+		m.logger.Debugf("retrying GetOrCreateStacktraces with %d stacktraces", len(stacktraceKeys))
+	}
+
+	if len(result.retryWith) != 0 {
+		m.logger.Warningf("failed to GetOrCreateStacktraces %d stacktraces", len(stacktraceKeys))
+		return res, fmt.Errorf("partial commit of stacktraces: %w", badger.ErrTxnTooBig)
```

Comment:
I'm going to wrap it in the end because we can infer the error and I didn't want to lose the info in the log that originally highlighted it